examples/karpenter-fargate/main.tf

```diff
@@ -0,0 +1,327 @@
+        name  = "podDisruptionBudget.maxUnavailable"
+        value = 1
+      }
+    ]
+  }
+
+  tags = local.tags
+}
+
+# # Allow ingress from the worker nodes security group (Karpenter nodes)
+# # to the cluster primary security group (Fargate nodes)
+# resource "aws_security_group_rule" "cluster_primary_ingress_all" {
+#   description              = "Allow All ingress from the worker nodes security group"
+#   type                     = "ingress"
+#   to_port                  = 0
+#   protocol                 = "-1"
+#   from_port                = 0
+#   security_group_id        = module.eks_blueprints.cluster_primary_security_group_id
+#   source_security_group_id = module.eks_blueprints.worker_node_security_group_id
+# }
+
+# Add the Karpenter Provisioners IAM Role
+# https://karpenter.sh/v0.19.0/getting-started/getting-started-with-terraform/#create-the-karpentercontroller-iam-role
+resource "aws_iam_role" "karpenter" {
+  name = "${local.name}-karpenter-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Sid    = ""
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        }
+      },
+    ]
+  })
+}
+
+data "aws_iam_policy" "eks_cni" {
+  arn = "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy"
+}
+
+data "aws_iam_policy" "eks_worker_node" {
+  arn = "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy"
+}
+
+data "aws_iam_policy" "ecr_read_only" {
+  arn = "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly"
+}
+
+data "aws_iam_policy" "instance_core" {
+  arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+}
+
+resource "aws_iam_role_policy_attachment" "karpenter_eks_cni" {
+  role       = aws_iam_role.karpenter.name
+  policy_arn = data.aws_iam_policy.eks_cni.arn
+}
+
+resource "aws_iam_role_policy_attachment" "karpenter_eks_worker_node" {
+  role       = aws_iam_role.karpenter.name
+  policy_arn = data.aws_iam_policy.eks_worker_node.arn
+}
+
+resource "aws_iam_role_policy_attachment" "karpenter_ecr_read_only" {
+  role       = aws_iam_role.karpenter.name
+  policy_arn = data.aws_iam_policy.ecr_read_only.arn
+}
+
+resource "aws_iam_role_policy_attachment" "karpenter_instance_core" {
+  role       = aws_iam_role.karpenter.name
+  policy_arn = data.aws_iam_policy.instance_core.arn
+}
+
+resource "aws_iam_instance_profile" "karpenter" {
+  name = "${local.name}-karpenter-instance-profile"
+  role = aws_iam_role.karpenter.name
+}
+
+# Add the default provisioner for Karpenter autoscaler
+data "kubectl_path_documents" "karpenter_provisioners" {
+  pattern = "${path.module}/provisioners/default_provisioner*.yaml"
+  vars = {
+    azs                     = join(",", local.azs)
+    iam-instance-profile-id = "${local.name}-karpenter-instance-profile"
+    eks-cluster-id          = local.name
+    eks-vpc_name            = local.name
+  }
+}
+
+resource "kubectl_manifest" "karpenter_provisioner" {
```

Comment:
Warning here, if you update a provisioner in futur terraform apply, it will first delete, then recreate it.
But when it delete it, it will delete all karpenter nodes associated to it..

Reply:
Yes you're right. The provisioner that was deployed in https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples/karpenter was also using this method but it was removed in https://github.com/aws-ia/terraform-aws-eks-blueprints/commit/89159115229738bc3c0a568547167f87aeef1480.